database/storage.go

```diff
@@ -137,9 +126,54 @@ func (s *Storage) rows(table string) (*sql.Rows, error) {
 }
 
 func (s *Storage) nRows(table string, n string) (*sql.Rows, error) {
-	nRows, err := s.conn.Query(fmt.Sprintf("SELECT * FROM `%s` LIMIT %s", table, n))
+	column, err := s.primaryColumn(table)
+	if err != nil {
+		return nil, err
+	}
+	nRows, err := s.conn.Query(fmt.Sprintf("SELECT * FROM `%s` ORDER BY %s DESC LIMIT %s", table, column, n))
 	if err != nil {
 		return nRows, err
 	}
 	return nRows, nil
 }
+
+// database returns the name of the database.
+func (s *Storage) database() (string, error) {
```

Comment:
I would recommend to avoid to use something that can be related to a private method for a class in OOP. for example injecting the driver here will give the same functionality but improving readability and reusability. Only my 2 cents :)

Reply:
@mauipipe Kindly SMTC? What driver would you recommend I inject?

Reply:
I guess he saying something in this line:
```go
func database(s Store) (string, error) {
  row := s.conn.QueryRow("SELECT @@hostname")
   var hostname string
   if err := row.Scan(&hostname); err != nil {
     return "", err
   }
   return hostname, nil
}
```

The main advantage here is dependency injection `database(s Store)` will make the code easier to test and compose, however if you apply this pattern you should go for it all the way down

Reply:
Ah nice. Thanks @rafaeljesus. I talked with @mauipipe yesterday and he explained that it's a matter of style. He also kindly shared some resources too. It looks cool. I will use it next time. This code base is now already legacy :)